### PR TITLE
[prometheus] Enable pulling images by digest

### PR DESCRIPTION
**Pull request by ivyostosh**
#### What this PR does / why we need it
Pulling images by digest allows users to “pin” an image to that version, and guarantee that the image is always the same ([read more](https://docs.docker.com/engine/reference/commandline/pull/#pull-an-image-by-digest-immutable-identifier)). This is not currently supported by the helm chart.

#### Which issue this PR fixes

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

- fixes #

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/prometheus/values.yaml [hunk 1]
```diff
@@ -107,6 +109,8 @@ server:
     repository: quay.io/prometheus/prometheus
     # if not set appVersion field from Chart.yaml is used
     tag: ""
+    # When digest is set to a non-empty value, images will be pulled by digest (regardless of tag value).
+    # digest: ""
```
Comment (monotek) [inline charts/prometheus/values.yaml:113]:
```suggestion
    digest: ""
```

File: charts/prometheus/values.yaml [hunk 2]
```diff
@@ -33,6 +33,8 @@ configmapReload:
     image:
       repository: jimmidyson/configmap-reload
       tag: v0.8.0
+      # When digest is set to a non-empty value, images will be pulled by digest (regardless of tag value).
+      # digest: ""
```
Comment (monotek) [inline charts/prometheus/values.yaml:37]:
```suggestion
      digest: ""
```

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/prometheus/Chart.yaml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v2
 name: prometheus
-appVersion: v2.40.5
-version: 19.0.3
+appVersion: v2.41.0
+version: 19.1.1
```
Comment (monotek) [inline charts/prometheus/Chart.yaml:4]:
```suggestion
version: 19.1.0
```

**Review by monotek — CHANGES_REQUESTED** [review]

**Review by monotek — CHANGES_REQUESTED** [review]
Sorry, Version needs to be 19.2.0 now...

**Comment by ivyostosh** [comment]
> Sorry, Version needs to be 19.2.0 now...

@monotek Updated!

**Review by monotek — APPROVED** [review]

**Review by zanhsieh — APPROVED** [review]

**Pull request merged**